File: app/src/main/java/com/example/monappliimmo/AppartementAdapter.java
package com.example.monappliimmo;

import android.content.Intent;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;
import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;
import java.util.List;

public class AppartementAdapter extends RecyclerView.Adapter<AppartementAdapter.AppartementViewHolder> {

    private List<Appartement> listeAppartements;

    public AppartementAdapter(List<Appartement> listeAppartements) {
        this.listeAppartements = listeAppartements;
    }

    @NonNull
    @Override
    public AppartementViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(parent.getContext())
                .inflate(R.layout.item_appartement, parent, false);
        return new AppartementViewHolder(view);
    }

    @Override
    public void onBindViewHolder(@NonNull AppartementViewHolder holder, int position) {
        Appartement appartement = listeAppartements.get(position);

        // Affichage dans la liste principale
        holder.tvNumero.setText("Appartement n°" + appartement.getNumero());
        holder.tvDescription.setText(appartement.getDescription());
        holder.tvSurface.setText(appartement.getSurface() + " m² - " + appartement.getNombrePieces() + " pièces");

        holder.itemView.setOnClickListener(v -> {
            Intent intent = new Intent(v.getContext(), DetailActivity.class);

            // 1. Données de l'objet Appartement
            intent.putExtra("ID", appartement.getId());
            intent.putExtra("NUMERO", String.valueOf(appartement.getNumero()));
            intent.putExtra("DESC", appartement.getDescription());
            intent.putExtra("SURFACE", appartement.getSurface());
            intent.putExtra("PIECES", appartement.getNombrePieces());

            // 2. RÉCUPÉRATION DU LOYER ET DU LOCATAIRE (Via le Contrat)
            // Dans ton SQL, ces infos sont dans la table 'contrat'
            if (appartement.getContrats() != null && !appartement.getContrats().isEmpty()) {
                // On récupère le premier contrat (le plus récent)
                Contrat contratActuel = appartement.getContrats().get(0);

                double totalLoyer = contratActuel.getMontantBrut() + contratActuel.getMontantCharge();
                intent.putExtra("PRIX", totalLoyer);

                if (contratActuel.getLocataire() != null) {
                    String nomComplet = contratActuel.getLocataire().getPrenom() + " " + contratActuel.getLocataire().getNom();
                    intent.putExtra("PROPRIO", nomComplet); // Ici on affiche le locataire actuel
                }
            } else {
                intent.putExtra("PRIX", 0.0);
                intent.putExtra("PROPRIO", "Logement vacant");
            }

            // 3. RÉCUPÉRATION DE L'ADRESSE (Via la jointure Batiment)
            if (appartement.getBatiment() != null) {
                String adresseComplete = appartement.getBatiment().getAdresse() + ", " + appartement.getBatiment().getVille();
                intent.putExtra("ADRESSE", adresseComplete);
            } else {
                intent.putExtra("ADRESSE", "Adresse non renseignée");
            }

            v.getContext().startActivity(intent);
        });
    }

    @Override
    public int getItemCount() {
        return (listeAppartements != null) ? listeAppartements.size() : 0;
    }

    public static class AppartementViewHolder extends RecyclerView.ViewHolder {
        TextView tvNumero, tvDescription, tvSurface;

        public AppartementViewHolder(@NonNull View itemView) {
            super(itemView);
            tvNumero = itemView.findViewById(R.id.textNumero);
            tvDescription = itemView.findViewById(R.id.textDescription);
            tvSurface = itemView.findViewById(R.id.textSurface);
        }
    }
}

File: app/src/main/java/com/example/monappliimmo/AppartementAdapter.kt
package com.example.monappliimmo

import android.content.Intent
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView

class AppartementAdapter(
    private val listeAppartements: List<Appartement>?,
    private val batimentParent: Batiment? = null
) : RecyclerView.Adapter<AppartementAdapter.AppartementViewHolder>() {

    constructor(listeAppartements: List<Appartement>?) : this(listeAppartements, null)

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): AppartementViewHolder {
        val view = LayoutInflater.from(parent.context)
            .inflate(R.layout.item_appartement, parent, false)
        return AppartementViewHolder(view)
    }

    override fun onBindViewHolder(holder: AppartementViewHolder, position: Int) {
        val appartement = listeAppartements?.get(position) ?: return

        holder.tvNumero.text = "Appartement n°${appartement.numero}"
        holder.tvDescription.text = appartement.description
        holder.tvSurface.text = "${appartement.surface} m² - ${appartement.nombrePieces} pièces"

        holder.itemView.setOnClickListener { v ->
            val context = v.context
            val intent = Intent(context, DetailActivity::class.java)

            intent.putExtra("ID", appartement.id)
            intent.putExtra("NUMERO", appartement.numero.toString())
            intent.putExtra("DESC", appartement.description)
            intent.putExtra("SURFACE", appartement.surface)
            intent.putExtra("PIECES", appartement.nombrePieces)

            val batLocal = appartement.batiment
            val adresseComplete = when {
                batimentParent != null -> "${batimentParent.adresse}, ${batimentParent.ville}"
                batLocal != null -> "${batLocal.adresse}, ${batLocal.ville}"
                else -> "Adresse non renseignée"
            }
            intent.putExtra("ADRESSE", adresseComplete)

            val contratActuel = appartement.contrats?.firstOrNull()

            if (contratActuel != null) {
                val totalLoyer = contratActuel.montantBrut + contratActuel.montantCharge
                intent.putExtra("PRIX", totalLoyer)

                val loc = contratActuel.locataire
                if (loc != null) {
                    val nomComplet = "${loc.prenom} ${loc.nom}"
                    intent.putExtra("PROPRIO", nomComplet)
                    intent.putExtra("LOCATAIRE_ID", loc.id)
                    intent.putExtra("LOCATAIRE_EMAIL", loc.mail)
                } else {
                    intent.putExtra("PROPRIO", "Locataire non renseigné")
                    intent.putExtra("LOCATAIRE_ID", -1L)
                }
            } else {
                intent.putExtra("PRIX", 0.0)
                intent.putExtra("PROPRIO", "Logement vacant")
                intent.putExtra("LOCATAIRE_ID", -1L)

            }

            context.startActivity(intent)
        }
    }

    override fun getItemCount(): Int {
        return listeAppartements?.size ?: 0
    }

    class AppartementViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
        val tvNumero: TextView = itemView.findViewById(R.id.textNumero)
        val tvDescription: TextView = itemView.findViewById(R.id.textDescription)
        val tvSurface: TextView = itemView.findViewById(R.id.textSurface)
    }
}